### PR TITLE
[DEM-727] Integrate DataSetIO into DataSet

### PR DESCRIPTION
**Pull request by qSaevar**

### CODE REVIEW

File: src/qilib/data_set/data_array.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class DataArray:
     """
 
     def __init__(self, name: str, label: str, unit: str = '', is_setpoint: bool = False,
-                 preset_data: Optional[numpy.ndarray] = None, set_arrays: Optional[List['DataArray']] = None,
+                 preset_data: Optional[numpy.ndarray] = None, set_arrays: Optional[Union[List['DataArray'], Tuple['DataArray', ...], 'DataArray']] = None,
```
Comment (ammar92) [inline src/qilib/data_set/data_array.py]: After the type hinting, this line looks pretty lengthy. Can we do something about it?

File: src/qilib/data_set/data_set_io_reader.py [hunk 2]
```diff
@@ -23,14 +23,19 @@
...
+        self._data_set: DataSet
 
-    @abstractmethod
     def bind_data_set(self, data_set: DataSet) -> None:
         """ Binds the DataSet to the DataSetIOReader.
 
         Args:
             data_set: The object that encompasses DataArrays.
         """
+        self._data_set = data_set
```
Comment (ammar92) [inline src/qilib/data_set/data_set_io_reader.py]: (Remark) My inspector warns about this attribute being defined outside of the constructor. We can ignore this until we find a solution for this.

File: src/qilib/data_set/data_array.py (hunk 1 above)
Comment (qSaevar) [reply src/qilib/data_set/data_array.py]: fix it, and, it's fixed

File: src/qilib/data_set/data_set_io_reader.py (hunk 2 above)
Comment (qSaevar) [reply src/qilib/data_set/data_set_io_reader.py]: ✔️

**Pull request merged**